NetCasbin/InternalEnforcer.cs

```diff
@@ -17,34 +17,27 @@ public class InternalEnforcer : CoreEnforcer
         /// <returns></returns>
         protected bool AddPolicy(string sec, string ptype, List<string> rule)
         {
-            var ruleAdded = model.AddPolicy(sec, ptype, rule);
-            if (!ruleAdded)
+            if (model.HasPolicy(sec, ptype, rule))
             {
                 return false;
-            } 
+            }
 
             if (adapter != null && autoSave)
             {
                 try
                 {
                     adapter.AddPolicy(sec, ptype, rule);
                 }
-                catch (NotImplementedException e)
-                {
-
-                }
-                catch (Exception e)
-                {
-                    throw e;
-                }
-
-                if (watcher != null)
+                catch (NotImplementedException)
                 {
                     // error intentionally ignored
-                    watcher.Update();
                 }
+
+                watcher?.Update();
```

Comment:
If the adapter does not update the policy,  it is unnecessary to call the watcher.